async_tcp_server.cpp

```diff
@@ -0,0 +1,101 @@
+#include "async_tcp_server.hpp"
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <sys/select.h>
+
+AsyncTCPServer::AsyncTCPServer() {}
+
+AsyncTCPServer::AsyncTCPServer(Socket_sptr master_socket, fd_set fdsetRead): m_fdsetRead(fdsetRead)
+{
+    FD_ZERO(&m_fdsetRead);
+    FD_SET(master_socket->GetSocketHandle(), &m_fdsetRead);
+    m_aSockets.push_back(master_socket);
+}
+
+Socket_sptr AsyncTCPServer::GetSocketByFD(int fd) const
+{
+    for(auto socket : m_aSockets)
+    {
+        if (socket->GetSocketHandle() == fd)
+            return socket;
+    }
+
+    return Socket_sptr(nullptr);
+}
+
+AsyncTCPServer::~AsyncTCPServer()
+{
+    try
+    {
+        GetMasterSocket()->Close();
+    }
+    catch (std::runtime_error & re)
```

Comment:
Lets be more specific here.
We're waiting for SocketException here: if socket closing was failed, it's of course bad, but maybe not critical.
But any other exception here might be a good reason to crash our app - something really went wrong.
You still can disagree with me - my opinion is not The Only Right in this world ;) And it would be nice to discuss any details.